sympy/polys/domains/quotientring.py

```diff
@@ -28,7 +28,8 @@ def __init__(self, ring, data):
 
     def __str__(self):
         from sympy.printing.str import sstr
-        return sstr(self.data) + " + " + str(self.ring.base_ideal)
+        data = self.ring.ring.to_sympy(self.data)
+        return sstr(data) + " + " + str(self.ring.base_ideal)
```

Comment:
In some cases the old printing behaviour can be recovered at a higher-level by using the ring to convert the `DMP` into an expression with symbols.